examples/chip-tool/commands/tests/TestCommand.h

```diff
@@ -92,6 +92,15 @@ class TestCommand : public CHIPCommand
 
         return true;
     }
+    template <typename T, typename U>
+    bool CheckConstraintMinValue(const char * itemName, const chip::app::DataModel::Nullable<T> & current, U expected)
+    {
+        if (current.IsNull())
+        {
+            return true;
```

Comment:
would we not  need to verify that we "expect null"?

The way I read it, this is: constraint validates to  true regardless of expected value as long as the value is null. That sounds off.

Reply:
Well there is the "value: null" keyword to test explicitly you want this value to be Null

Here in case the value is not set we want to basically skip the test 

Then you might want to test the null explicitly i another step or condition

the way the  YAML's syntax work right now this is the best to do

Reply:
Here how it work with value   
 - label: "Write attribute NULLABLE_INT8U Max Value"
      command: "writeAttribute"
      attribute: "nullable_int8u"
      arguments:
          value: 254

    - label: "Read attribute NULLABLE_INT8U Max Value"
      command: "readAttribute"
      attribute: "nullable_int8u"
      response:
          value: 254

    - label: "Write attribute NULLABLE_INT8U Invalid Value"
      command: "writeAttribute"
      attribute: "nullable_int8u"
      arguments:
          value: 255
      response:
          error: CONSTRAINT_ERROR

    - label: "Read attribute NULLABLE_INT8U unchanged Value"
      command: "readAttribute"
      attribute: "nullable_int8u"
      response:
          value: 254

    - label: "Write attribute NULLABLE_INT8U null Value"
      command: "writeAttribute"
      attribute: "nullable_int8u"
      arguments:
          value: null

    - label: "Read attribute NULLABLE_INT8U null Value"
      command: "readAttribute"
      attribute: "nullable_int8u"
      response:
          value: null

Reply:
If I read a test of "expect value to be 100" then I would expect it to pass IFF the value is 100, not to pass if 100 or null.

Can't we have a 'or null' qualifier somehow instead? It is not obvious that "missing" is valid for any value.

Reply:
Imagine this test:

  - write value 100
  - read back value, expect 100
  
if we allow null to always pass, this passes when the write fails to store the value (because read will return null, which matches 100)

Reply:
this is not the case here since it is about checking IF the value is set , value has to be within a valid range defined between MIN and MAX

Reply:
> Imagine this test:
> 
> * write value 100
> * read back value, expect 100
> 
> if we allow null to always pass, this passes when the write fails to store the value (because read will return null, which matches 100)

You are right indeed and the tool are present for this ! here we want to fix the fact that an existent value is withing a valid range only or not a value .

Reply:
Please add a comment saying that for min/max range checks we explicitly allow null.

It seems unfortunate to me that I cannot say "value MUST be between a and b" without allowing null. Hoping that there are other tests that say "must not be null"

Reply:
@andy31415 For what it's worth, `null` in the spec was precisely introduced in a way so you could say "value between 3 and 17" in the table and have that mean "value is between 3 and 17 or is null".  So in terms of certification, we need a way to test for the spec-required behavior, which in fact treats null as in-range.  That's the behavior this is implementing.

We may want to add an isNotNull constraint so you could write:
```
  minValue: 5
  maxValue: 17
  isNotNull: true
```
or we could make this work:
```
  minValue: 5
  maxValue: 17
  notValue: null
```
using the existing `notValue` constraint.  I mildly prefer the latter, fwiw.